library/Faktory/Worker.hs

```diff
@@ -106,6 +111,12 @@ processorLoop client settings workerSettings f = do
                     failJob client job $ T.pack $ show ex
                   ]
 
+getJobTTL :: Job arg -> Int
+getJobTTL = (* 1000) . maybe faktoryTimeoutDefault fromIntegral . jobReserveFor
```

Comment:
This is actually what I would have `jobReserveFor` be defined as, personally. Using `joReserveFor` in the JSON encoding is a little related to this. I started down this path of `Job`-level options accessors being a good seam to add defaulting with `jobRemainingRetries`, for example.

```suggestion
jobReserveFor = (* 1000) . maybe faktoryTimeoutDefault fromIntegral . joReserveFor . jobOptions
```